cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/QueryableDatastore.java

```diff
@@ -0,0 +1,282 @@
+package com.cloudant.sync.datastore;
+
+import com.cloudant.sync.datastore.encryption.KeyProvider;
+import com.cloudant.sync.datastore.encryption.NullKeyProvider;
+import com.cloudant.sync.datastore.migrations.SchemaOnlyMigration;
+import com.cloudant.sync.query.CheckedQueryException;
+import com.cloudant.sync.query.Index;
+import com.cloudant.sync.query.IndexCreator;
+import com.cloudant.sync.query.IndexType;
+import com.cloudant.sync.query.IndexUpdater;
+import com.cloudant.sync.query.QueryConstants;
+import com.cloudant.sync.query.QueryExecutor;
+import com.cloudant.sync.query.QueryResult;
+import com.cloudant.sync.sqlite.Cursor;
+import com.cloudant.sync.sqlite.SQLDatabase;
+import com.cloudant.sync.sqlite.SQLDatabaseQueue;
+import com.cloudant.sync.sqlite.SQLQueueCallable;
+import com.cloudant.sync.util.DatabaseUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+/**
+ * Created by rhys on 04/07/2016.
+ */
+public class QueryableDatastore extends DatastoreImpl {
+
+
+    private static final String FTS_CHECK_TABLE_NAME = "_t_cloudant_sync_query_fts_check";
+
+
+    private static final String EXTENSION_NAME = "com.cloudant.sync.query";
+    private static final String INDEX_FIELD_NAME_PATTERN = "^[a-zA-Z][a-zA-Z0-9_]*$";
+
+    private static final Logger logger = Logger.getLogger(QueryableDatastore.class.getName());
+
+    private final Pattern validFieldName;
+
+    private final SQLDatabaseQueue dbQueue;
```

Comment:
I think this should be renamed to make it clear that this `dbQueue` is for the index databases, not the datastore that this extends.